packages/datadog-instrumentations/src/openai.js

```diff
@@ -144,56 +144,57 @@ addHook({ name: 'openai', file: 'dist/api.js', versions: ['>=3.0.0 <4'] }, expor
 })
 
 function addStreamedChunk (content, chunk) {
-  return content.choices.map((oldChoice, choiceIdx) => {
-    const newChoice = oldChoice
-    const chunkChoice = chunk.choices[choiceIdx]
-    if (!oldChoice.finish_reason) {
-      newChoice.finish_reason = chunkChoice.finish_reason
-    }
+  for (const choice of chunk.choices) {
+    const choiceIdx = choice.index
+    if (!content.choices.some(choice => choice.index === choiceIdx)) {
+      content.choices.push(choice)
```

Comment:
I actually want to make this a bit more dynamic; I think right now, although unlikely, if chunks have indices out of order, then accessing below with `content.choices[choiceIdx]` won't work if we push a choice with `index == 3` as the first choice. Although I don't think this happens from the OpenAI side, I'll add some defensive logic here just in case.